core/src/main/clojure/xtdb/node/impl.clj

```diff
@@ -183,7 +190,8 @@
 (defmethod ig/init-key :xtdb/node [_ {:keys [metrics-registry] :as deps}]
   (let [node (map->Node (-> deps
                             (assoc :query-timer (metrics/add-timer metrics-registry "query.timer"
-                                                                   {:description "indicates the timings for queries"}))))]
+                                                                   {:description "indicates the timings for queries"}))
```

Comment:
Let's add the `warning` counter on node level. On the `PreparedQuery` object you can call the `warnings` method.